src/Microsoft.Health.Dicom.Core/Features/Telemetry/RetrieveMeter.cs

```diff
@@ -0,0 +1,25 @@
+// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Diagnostics.Metrics;
+
+namespace Microsoft.Health.Dicom.Core.Features.Telemetry;
+
+public sealed class RetrieveMeter : IDisposable
+{
+    private readonly Meter _meter;
+
+    public RetrieveMeter()
+    {
+        _meter = new Meter($"{OpenTelemetryLabels.BaseMeterName}.Instance", "1.0");
+        RetrieveInstanceCount = _meter.CreateCounter<long>(nameof(RetrieveInstanceCount), description: "Total number of instances retrieved");
```

Comment:
I think you can leave `Retrieve` out of the name given it's in the `RetrieveMeterCount`, unless that context isn't obvious downstream. How does this metric appear in App Insights?

Reply:
Meter name doesnt come in App insight, only metrics name will. I think it's best to keep it as RetrieveInstanceCount.